### PR TITLE
small fixes for macos

### LINKED ISSUE #510: [APPLE]Heap corruption when loading Cubemap w/mipmaps
with automaticRendering on on mac and validation tests, I got a recurring crash with napi here: https://github.com/BabylonJS/BabylonNative/blob/cb2f58449157f78c56109787d247c67f732e84ed/Plugins/NativeEngine/Source/NativeEngine.cpp#L1162
I recoded the lambda to something like this:
```
                auto* pointer = static_cast<uint8_t*>(typedArray.ArrayBuffer().Data()) + typedArray.ByteOffset();
                std::vector<uint8_t> memory(pointer, pointer + typedArray.ByteLength());
                tasks[(face * numMips) + mip] = arcana::make_task(arcana::threadpool_scheduler, m_cancelSource, [this, memory = std::move(memory)]() {
                    bimg::ImageContainer* image = bimg::imageParse(&m_allocator, memory.data(), memory.size());
                    FlipY(image);
                    return image;
                });
```
And crash disappeared.
I wanted to make a capture of the callstack of the crash for this issue but once my code discarded, I get random crash with heap corruption.
At this point, I don't know if the code is the culprit or moving/replacing code will hide the memory corruption.

I finally managed to get a callstack with the crash in napi:
image

### PR DESCRIPTION
**Pull request by CedricGuillemet**
- fix URL containing forbiden characters
- bool RENDER_ON_JS_THREAD on macOS validationtests

### CODE REVIEW

File: Dependencies/UrlLib/Source/Apple/UrlRequest.mm [hunk 1]
```diff
@@ -56,6 +58,11 @@ void ResponseType(UrlResponseType value)
 
             NSURLSession* session{[NSURLSession sharedSession]};
             NSURLRequest* request{[NSURLRequest requestWithURL:url]};
+            if (url == nil)
+            {
+                // Complete the task, but retain the default status code of 0 to indicate a client side error.
+                return arcana::task_from_result<std::exception_ptr>();
```
Comment (bghgary) [inline Dependencies/UrlLib/Source/Apple/UrlRequest.mm:64]: We still need to work on getting the status text propagated back out for this to be fully correct.

File: Apps/ValidationTests/macOS/ViewController.mm [hunk 2]
```diff
@@ -75,7 +76,7 @@ - (void)refreshBabylon {
         Babylon::Polyfills::XMLHttpRequest::Initialize(env);
 
         graphics->AddToJavaScript(env);
-        Babylon::Plugins::NativeEngine::Initialize(env, false); // render on UI Thread
+        Babylon::Plugins::NativeEngine::Initialize(env, RENDER_ON_JS_THREAD);
```
Comment (bghgary) [inline Apps/ValidationTests/macOS/ViewController.mm:79]: We are now changing this from false to true. What is the reason why it doesn't work?

**Review by bghgary — APPROVED** [review]
Okay to merge, but I have some comments/questions.

File: Apps/ValidationTests/macOS/ViewController.mm (hunk 2 above)
Comment (CedricGuillemet) [reply Apps/ValidationTests/macOS/ViewController.mm:79]: Because of this : https://github.com/BabylonJS/BabylonNative/issues/512

Comment (bghgary) [reply Apps/ValidationTests/macOS/ViewController.mm:79]: 👍

**Pull request merged**